tests/testthat/test-with.R

```diff
@@ -0,0 +1,29 @@
+test_that("with_defaults works as expected with unnamed args", {
+  # assignment_linter is in defaults, so output doesn't change
+  expect_named(with_defaults(assignment_linter), names(with_defaults()))
+})
+
+test_that("all default linters are tagged default", {
+  expect_named(with_defaults(), available_linters(tags = "default")$linter)
+
+  # TODO(michaelchirico): use plain expect_equal after waldo#133 makes it into a CRAN release
+  # Here, the environment()s are different because factories use them.
+  skip_if_not(getRversion() >= "4.1") # Desired all.equal behaviour only available in >= 4.1
+  # covr modifies package functions causing differing deparse() results even for identical anonymous functions.
```

Comment:
let's file a bug & reference it here to eventually remove it

Reply:
Not sure if covr can do much here because the `default_linters` list is prebuilt. If my suspicion is correct, `with_tags()[["assignment_linter"]]` should be equal to `assignment_lintr()` even in covr, because both are constructed at runtime.

Reply:
Elaborated the comment in hopes of making the dilemma clear. Not a covr bug in my view -- simply an artifact of how we define `default_linters`.

Reply:
i see, makes sense, thanks. agree with the skip.